src/calva-fmt/src/config.ts

```diff
@@ -11,7 +11,8 @@ function readConfiguration() {
         "insert-missing-whitespace?": workspaceConfig.get("insertMissingWhitespace"),
         "remove-consecutive-blank-lines?": workspaceConfig.get("removeConsecutiveBlankLines"),
         "align-associative?": workspaceConfig.get("alignMapItems"),
-        "cljfmt-edn": state.readConfigFile(workspaceConfig.get("configPath"))
+        "cljfmt-edn": state.readConfigFile(workspaceConfig.get("configPath")),
+        "cljfmt-parsed": state.readConfigEdn(workspaceConfig.get("configPath")),
```

Comment:
This looks like two configs reading the same setting. Am I reading it incorrectly, or is it some transitive wip-thing? 😄

Reply:
you are right, it is WIP. I am struggling to find a good way to work with configuration file.

* the indentation rules need map with indents `cljfmt-parsed` in src/cursor-doc/indent.ts
* the formatter in cljs need just string, otherwise we loose type information `cljfmt-edn` in src/cljs-lib/src/calva/fmt/formatter.cljs
* cache content of config file to avoid reading it on every keypress

Maybe you have some opinions or thoughts about this?

Reply:
I've updated the code. Could you please have a look if it makes sense?

Reply:
Maybe we should use `workspace.createFileSystemWatcher()` somehow, and keep the formatting config updated that way?

Reply:
`workspace.createFileSystemWatcher()` should work, but it has one limitation - it works only for files in workspace. It means that users will not be use a file from somewhere in home folder.

Reply:
Yeah, there's that, of course... what's your opinion on not supporting that?

We use node file watchers elsewhere, but we are planning to change those to vscode ones, to better support Remote and such.

Reply:
Yes, it makes sense. From my small experience the Remote is more valuable than config file external to the project. I've just tested that user can use symlink as a workaround.